tests/db/test_tracking_operations.py

```diff
@@ -41,3 +45,97 @@ def test_set_run_status_to_killed():
+    # SQLAlchemy OperationalError. PEP 249 describes OperationalError as:
+    #
+    # > Exception raised for errors that are related to the database’s operation
+    # > and not necessarily under the control of the programmer, e.g. an
+    # > unexpected disconnect occurs, the data source name is not found, a
+    # > transaction could not be processed, a memory allocation error occurred
+    # > during processing, etc.
+    #
+    # These errors are typically transient and can be resolved by retrying the
+    # operation, hence MLflow has different handling for them as compared to
+    # the more generic exception type, SQLAlchemyError.
+    #
+    # This is particularly important for REST clients, where
+    # TEMPORARILY_UNAVAILABLE triggers MLflow REST clients to retry the request,
+    # whereas BAD_REQUEST does not.
+    execute_counter = None
+    api_module = None
+    old_connect = None
+    old_dbapi = sqlalchemy.dialects.sqlite.pysqlite.SQLiteDialect_pysqlite.dbapi
+
+    class ConnectionWrapper:
+        """Wraps a sqlite3.Connection object."""
+
+        def __init__(self, conn):
+            self.conn = conn
+
+        def __getattr__(self, name):
+            return getattr(self.conn, name)
+
+        def cursor(self):
+            """Return a wrapped SQLite cursor."""
+            return CursorWrapper(self.conn.cursor())
+
+    class CursorWrapper:
+        """Wraps a sqlite3.Cursor object."""
+
+        def __init__(self, cursor):
+            self.cursor = cursor
+
+        def __getattr__(self, name):
+            return getattr(self.cursor, name)
+
+        def execute(self, *args, **kwargs):
+            """Wraps execute(), simulating sporadic OperationalErrors."""
+            nonlocal execute_counter
+            if execute_counter is not None and "pragma" not in args[0].lower():
+                execute_counter += 1
+                if execute_counter % 3 == 0:
+                    # Simulate a database error
+                    raise sqlite3.OperationalError("test")
+            return self.cursor.execute(*args, **kwargs)
+
+    def connect(*args, **kwargs):
+        """Wraps sqlite3.dbapi.connect(), returning a wrapped connection."""
+        global connect_counter
+        conn = old_connect(*args, **kwargs)
+        return ConnectionWrapper(conn)
+
+    def dbapi(*args, **kwargs):
+        """Wraps SQLiteDialect_pysqlite.dbapi(), returning patched dbapi."""
+        nonlocal api_module, old_connect
+        if api_module is None:
+            # Only patch the first time dbapi() is called, to avoid recursion.
+            api_module = old_dbapi(*args, **kwargs)
+            old_connect = api_module.connect
+            monkeypatch.setattr(api_module, "connect", connect)
+        return api_module
+
+    monkeypatch.setattr(sqlalchemy.dialects.sqlite.pysqlite.SQLiteDialect_pysqlite, "dbapi", dbapi)
+
+    with pytest.raises(mlflow.MlflowException):
```

Comment:
```suggestion
    with pytest.raises(mlflow.MlflowException, match=...):
```

Can we add `match`? `pytest.raises(mlflow.MlflowException)` matches any `MlflowException`.

Reply:
Sure!